### PR TITLE
Add a test for intrinsic size checks

### PR DESCRIPTION
**Pull request by jiahaog**
## Description

Add a test to assert that when doing intrinsic size checks, the parent `RenderBox` returns the correct intrinsics.

## Related Issues

#70730

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/test/rendering/dynamic_intrinsics_test.dart [hunk 1]
```diff
@@ -83,4 +83,42 @@ void main() {
     pumpFrame();
     expect(root.size, equals(inner.size));
   });
+
+  group('When RenderObject.debugCheckingIntrinsics is true', () {
+    setUp(() => RenderObject.debugCheckingIntrinsics = true);
+    tearDown(() => RenderObject.debugCheckingIntrinsics = false);
```
Comment (Hixie) [inline packages/flutter/test/rendering/dynamic_intrinsics_test.dart:89]: can we do this inside the test? it's much clearer what is going on if we don't use `test`'s various features. (For example, the timing of `testWidgets`'s invariants check and `tearDown` isn't obvious.) That would also let you get rid of `group`.

**Review by Hixie — APPROVED** [review]
This is great, thanks!

LGTM

I assume with this your other PR ends up failing?

File: packages/flutter/test/rendering/dynamic_intrinsics_test.dart (hunk 1 above)
Comment (jiahaog) [reply packages/flutter/test/rendering/dynamic_intrinsics_test.dart]: Sure, I replaced this with a try-finally to reset `RenderObject.debugCheckingIntrinsics` and removed the `group`.

**Comment by jiahaog** [comment]
Yup, it fails with the #70730 patched on top. Thanks for catching it!

**Pull request merged**